src/logic/messages/EventManager.cpp

```diff
@@ -121,50 +172,27 @@ void EventManager::processMessageQueue()
     }
 }
 
-EventMessages::EventMessage* EventManager::getTalkingWithMessage(Handle::EntityHandle other)
+EventMessages::EventMessage* EventManager::findLastConvMessageWith(Handle::EntityHandle other)
 {
-    EventMessages::EventMessage* lastNonOverlay = nullptr;
-
-    for(EventMessages::EventMessage* ev : m_EventQueue)
-    {
-        if(!ev->isOverlay)
-        {
-            lastNonOverlay = ev;
-        }
-
-        if(ev->messageType == EventMessages::EventMessageType::Conversation)
+    auto begin = m_EventQueue.rbegin();
+    auto end = m_EventQueue.rend();
+    auto lastConvMessage = std::find_if(begin, end, [&](EventMessages::EventMessage* ev){
+        if(!ev->isOverlay && ev->messageType == EventMessages::EventMessageType::Conversation)
         {
             auto conv = reinterpret_cast<EventMessages::ConversationMessage*>(ev);
```

Comment:
I know this isn't your code, but since you're changing the surroundings, switch the cast to dynamic please.